Common/Securities/SecurityPortfolioManager.cs

```diff
@@ -96,7 +96,23 @@ public SecurityPortfolioManager(SecurityManager securityManager, SecurityTransac
             // default to $100,000.00
             _baseCurrencyCash.SetAmount(100000);
 
-            CashBook.Updated += (sender, args) => InvalidateTotalPortfolioValue();
+            CashBook.Updated += (sender, args) =>
+            {
+                if (args.UpdateType == CashBook.UpdateType.Added)
+                {
+                    // add the same currency entry to the unsettled cashbook as well
+                    UnsettledCashBook.Add(args.Cash.Symbol, new Cash(args.Cash.Symbol, 0, args.Cash.ConversionRate));
+                }
+                // if updated, let's keep track of conversion rate changes
+                else if (args.UpdateType == CashBook.UpdateType.Updated &&
```

Comment:
> CashBook.UpdateType

Think we can take advantage we are changing these files to rename the enum to `CashUpdateType` and push it to Global.cs, think we would be able to get rid off the `using static` too

> // if updated, let's keep track of conversion rate changes

Nice catch! This is an interesting bug. If the cash instances shared the same `CurrencyConversion` instance I think we would be able to get around the need for this. We would need to push `_conversionRateNeedsUpdate` into `ICurrencyConversion` implementation though